05-Mining-Protocol.md

```diff
@@ -252,10 +255,11 @@ Client sends result of its hashing work to the server.
 Only relevant for extended channels.
 The message is the same as `SubmitShares`, with the following additional field:
 
-| Field Name                              | Data Type | Description                                                                                                                                                                                                                                                                                |
-| --------------------------------------- | --------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Field Name                              | Data Type        | Description                                                                                                                                                                                                                                                                                                 |
+| --------------------------------------- |------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | `<SubmitSharesStandard message fields>` |
-| extranonce                              | B0_32     | Extranonce bytes which need to be added to coinbase to form a fully valid submission (full coinbase = coinbase_tx_prefix + extranonce_prefix + extranonce + coinbase_tx_suffix). The size of the provided extranonce MUST be equal to the negotiated extranonce size from channel opening. |
+| extranonce                              | B0_32            | Extranonce bytes which need to be added to coinbase to form a fully valid submission (full coinbase = coinbase_tx_prefix + extranonce_prefix + extranonce + coinbase_tx_suffix). The size of the provided extranonce MUST be equal to the negotiated extranonce size from channel opening.                  |
+| user_identity                           | OPTION[STR0_255] | Optional field, up to 32 bytes. The unique identity of the worker. This field will only be included if the `REQUIRES_EXTENDED_SHARES_IDENTIFICATION` flag is set in the `SetupConnection` message. If supported by the server, this feature allows for tracking the worker's specific hashrate server side. |
```

Comment:
There is a bit confusion. This field should not be OPTION.

If the identification is supposed to be negotiated in setup-connection phase, then this field should be mandatory.
Otherwise it shouldn't be there **at all**.

Don't forget that empty Option still takes one byte. So this in fact is different kind of option than the one listed in https://github.com/stratum-mining/sv2-spec/blob/main/03-Protocol-Overview.md#31-data-types-mapping